pkg/apis/core/v1alpha1/types_shoot.go

```diff
@@ -1279,6 +1279,9 @@ type WorkerKubernetes struct {
 type Machine struct {
 	// Type is the machine type of the worker group.
 	Type string `json:"type" protobuf:"bytes,1,opt,name=type"`
+	// Architecture is the CPU architecture of the machines in this worker pool.
+	// +optional
+	Architecture *string `json:"architecture,omitempty" protobuf:"bytes,20,opt,name=architecture"`
```

Comment:
- new fields new to be added to the end of the struct
- this field needs to be cross-checked against the cloudprofile, whether the machine type and image support the specified archictecture
- so probably move this change to a dedicated PR that adds the validation once https://github.com/gardener/gardener/pull/6178 is merged